ui/apps/platform/src/Containers/NetworkGraph/simulation/NetworkPoliciesGenerationScope.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import { Flex } from '@patternfly/react-core';
+import uniq from 'lodash/uniq';
+
+import { ClusterIcon, DeploymentIcon, NamespaceIcon } from '../common/NetworkGraphIcons';
+
+import './NetworkPoliciesGenerationScope.css';
+
+export type EntityScope = {
+    // `granularity` refers to the most specific entity type that has been selected by the user.
+    granularity: 'CLUSTER' | 'NAMESPACE' | 'DEPLOYMENT';
+    cluster: string;
+    namespaces: string[];
+    deployments: {
+        namespace: string;
+        name: string;
+    }[];
+};
+
+export type NetworkPoliciesGenerationScopeProps = {
+    networkPolicyGenerationScope: EntityScope;
+};
+
+function NetworkPoliciesGenerationScope({
+    networkPolicyGenerationScope,
+}: NetworkPoliciesGenerationScopeProps) {
+    const { granularity, cluster, deployments } = networkPolicyGenerationScope;
+
+    let deploymentElement = <span>All deployments</span>;
+    let namespaceElement = <span>All namespaces</span>;
+
+    if (granularity === 'NAMESPACE' || granularity === 'DEPLOYMENT') {
+        const namespaces = uniq(deployments.map((deployment) => deployment.namespace));
+        const namespaceCount = namespaces.length;
+        const namespaceText =
+            namespaceCount === 1 ? `namespace "${namespaces[0]}"` : `${namespaceCount} namespaces`;
+        namespaceElement = <span>{namespaceText}</span>;
+    }
+
+    if (granularity === 'DEPLOYMENT') {
+        const deploymentCount = deployments.length;
+        const deploymentText =
+            deploymentCount === 1
+                ? `deployment "${deployments[0].name}"`
```

Comment:
same here in `deployment blah` vs `blah`. especially since we already have the icons for each of the scope levels